src/crypto.ts

```diff
@@ -166,12 +166,15 @@ export function hashDataKey(datakey: string): Uint8Array {
  * @param registryEntry - Registry entry to hash.
  * @returns - Hash of the registry entry.
  */
-export function hashRegistryEntry(registryEntry: RegistryEntry): Uint8Array {
-  return hashAll(
-    hashDataKey(registryEntry.datakey),
-    encodeString(registryEntry.data),
-    encodeBigintAsUint64(registryEntry.revision)
-  );
+export function hashRegistryEntry(registryEntry: RegistryEntry, hashedDataKeyHex: boolean): Uint8Array {
```

Comment:
missing `@param`, docstring could use a comment explaining what it does exactly or how it influences the behaviour of the function